### PR TITLE
Optimize zoom out implementation with separate padding kernel 

### PR DESCRIPTION
**Pull request by chirayuG-nvidia**
Optimizing zoom-out image transform with work efficient padding kernel. Changes are simple on python side, existing unit tests will suffice and are passing.

CC: @shekhardw @gigony

### CODE REVIEW

**Comment by shekhardw** [comment]
@chirayuG-nvidia 
Changes look good to me. Approving from my side. It will be good if you can add test case for zoom out. Current test case only covers zoom in.

**Comment by chirayuG-nvidia** [comment]
@jakirkham can you please review and merge this PR?

File: python/cucim/src/cucim/core/operations/intensity/kernel/cuda_kernel_source.py [hunk 1]
```diff
@@ -169,74 +169,71 @@
...
+                                  int no_padding_h_end, int no_padding_w_end) {
+  // H -> block Y, row
+  // W -> block X, col
 
-    // corner replication not very friendly if large area to patch -
-    // single thread issues stores
-    // ToDo: Consider adding another kernel for corner padding
+  int out_pixel_h = blockIdx.y * blockDim.y + threadIdx.y;
+  int out_pixel_w = blockIdx.x * blockDim.x + threadIdx.x;
+  int image_start_offset = blockIdx.z * pitch;
```
Comment (grlee77) [inline python/cucim/src/cucim/core/operations/intensity/kernel/cuda_kernel_source.py:184]: can either remove this `image_start_offset` or use it below

**Review by grlee77 — APPROVED** [review]
Looks good to me. I notice one apparently unused variable in the new kernel

File: python/cucim/src/cucim/core/operations/intensity/kernel/cuda_kernel_source.py (hunk 1 above)
Comment (chirayuG-nvidia) [reply python/cucim/src/cucim/core/operations/intensity/kernel/cuda_kernel_source.py]: Removed.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
@gpucibot merge

**Pull request merged**